extensions/ql-vscode/src/run-queries-shared.ts

```diff
@@ -0,0 +1,596 @@
+  }
+
+  get evalLogEndSummaryPath() {
+    return findQueryEvalLogEndSummaryFile(this.querySaveDir);
+  }
+
+  get resultsPaths() {
+    return {
+      resultsPath: path.join(this.querySaveDir, 'results.bqrs'),
+      interpretedResultsPath: path.join(this.querySaveDir,
+        this.metadata?.kind === 'graph'
+          ? 'graphResults'
+          : 'interpretedResults.sarif'
+      ),
+    };
+  }
+  getSortedResultSetPath(resultSetName: string) {
+    return path.join(this.querySaveDir, `sortedResults-${resultSetName}.bqrs`);
+  }
+
+  /**
+   * Creates a file in the query directory that indicates when this query was created.
+   * This is important for keeping track of when queries should be removed.
+   */
+  async createTimestampFile() {
+    await createTimestampFile(this.querySaveDir);
+  }
+
+
+  /**
+   * Holds if this query can in principle produce interpreted results.
+   */
+  canHaveInterpretedResults(): boolean {
+    if (!this.databaseHasMetadataFile) {
+      void logger.log('Cannot produce interpreted results since the database does not have a .dbinfo or codeql-database.yml file.');
+      return false;
+    }
+
+    const kind = this.metadata?.kind;
+    const hasKind = !!kind;
+    if (!hasKind) {
+      void logger.log('Cannot produce interpreted results since the query does not have @kind metadata.');
+      return false;
+    }
+
+    // Graph queries only return interpreted results if we are in canary mode.
+    if (kind === 'graph') {
+      return config.isCanary();
+    }
+
+    // table is the default query kind. It does not produce interpreted results.
+    // any query kind that is not table can, in principle, produce interpreted results.
+    return kind !== 'table';
+  }
+
+  /**
+   * Holds if this query actually has produced interpreted results.
+   */
+  async hasInterpretedResults(): Promise<boolean> {
+    return fs.pathExists(this.resultsPaths.interpretedResultsPath);
+  }
+
+  /**
+   * Holds if this query already has DIL produced
+   */
+  async hasDil(): Promise<boolean> {
+    return fs.pathExists(this.dilPath);
+  }
+
+  /**
+   * Holds if this query already has CSV results produced
+   */
+  async hasCsv(): Promise<boolean> {
+    return fs.pathExists(this.csvPath);
+  }
+
+  /**
+   * Returns the path to the DIL file produced by this query. If the query has not yet produced DIL,
+   * this will return first create the DIL file and then return the path to the DIL file.
+   */
+  async ensureDilPath(cliServer: CodeQLCliServer): Promise<string> {
+    if (await this.hasDil()) {
+      return this.dilPath;
+    }
+    const compiledQuery = path.join(this.querySaveDir, 'compiledQuery.qlo');
+    if (!(await fs.pathExists(compiledQuery))) {
+      if (await cliServer.cliConstraints.supportsNewQueryServer()) {
+        // This could be from the new query server
+        // in whcih case we expect the qlo to be missing so we should ignore it
+        throw new Error(
```

Comment:
When would this happen? Is there anything the user can do to fix this situation? The error message as it is is not very helpful.

Reply:
Originally I though that this should not happen unless the user deleted the dil or a bug happened in the query server. But I have also realised that a compilation error would also cause this. It is a bit tricky to work out which case but hopefully it is understandable that if your query doesn't compile then you won't be able to see the DIL.